docs/developing-operators.md

```diff
@@ -84,6 +84,75 @@ Now your first operator is ready and you can install it to cluster. You can do i
+
+```yaml
+name: operator
+description: my first super awesome operator
+version: "5.7"
+kudoVersion: ">= 0.2.0"
+kubeVersion: ">= 1.14"
+maintainers:
+  - Bob <bob@example.com>
+  - Alice <alice@example.com>
+url: https://github.com/myoperator/myoperator
+```
+
+Most of these are provided as a form of documentation. `kudoVersion` and `kubeVersion` use semver constraints to define minimal or maximal version of kubernetes or kudo that this operator supports. Under the hood, we use [this library](https://github.com/Masterminds/semver) to evaluate the constraints.
+
+### Tasks section
+
+Another part of `operator.yaml` is tasks section. Tasks are the smallest pieces of work that get executed together. You usually group kubernetes manifests that should be applied at once into one task. An example can be a deploy task that will result in config.yaml and pod.yaml being applied to your cluster.
+
+```yaml
+tasks:
+  deploy-task:
+    resources:
+      - config.yaml
+      - pod.yaml
+```
+
+### Plans section
+
+Plans orchestrate tasks through `phases` and `steps`.
+
+Each Plan is a tree with a fixed three-level hierarchy of the plan itself, its phases, and then steps within those phases. The choice of three levels was arbitrarily chosen as “enough levels for anybody”. This three-level hierarchy can look as follows:
+
+```text
+Plan foo
+├─ Phase bar
+│  ├─ Step qux
+│  └─ Step quux
+└─ Phase baz
+   ├─ Step quuz
+   ├─ Step corge
+   └─ Step grault
+```
+
+
+Plans consists of one or more `phases`. `Phases` consists of one or more `steps`. `Steps` contain one or more `tasks` (those are defined in the section we talked about in the last paragraph). Both phases and also steps can be configured with an execution `strategy`, either `serial` or `parallel`.
+
+The sample has a `deploy` plan with a `deploy-phase` and a `deploy-step`. From the `deploy-step` the `deploy-task` is referenced. This task gets executed when an instance is created using the operator.
+
+At the same time, `deploy` plan is the most important plan within your operator because that is the default plan that every operator has to have and also plan that gets executed when you install instance of your operator into the cluster. Another important plans that you might consider having is `update` (run when instance metadata is updated) or `upgrade` (run when instance is upgraded from one version of oprator to another). If you don't provide `update` and/or `upgrade` plans for your operator, the fallback is always deploy.
+ 
+```yaml
+plans:
+  deploy:
+    strategy: serial
+    phases:
+      - name: deploy-phase
+        strategy: parallel
+        steps:
+          - name: deploy-step
+            tasks:
+              - deploy-task
+```
+
+Plans allow operators to see what the operator is currently doing, and to visualize the broader operation such as for a config rollout. The fixed structure of that information meanwhile makes it straightforward to build UIs and tooling on top.
+
```

Comment:
this is a good place to show `kudo plan status` output

Reply:
I am not sure that should be in developing operators :/ maybe we need another portion of the docs for running that operator and then seeing how the plan is executing.

Reply:
looking at the issues, that is definitely something I would expect in here https://github.com/kudobuilder/kudo/issues/395 ... so basically here you create your operator and in that other docs you actually tell people how to run it and monitor it. We could link that other doc from the end of this doc :) that would be my proposal